src/test/resources/regressions/features/ghost_pointer/ghost-pointer-receiver-fail01.gobra

```diff
@@ -0,0 +1,149 @@
+    (*p).GhostPointerReceiver() // fails because method expects a ghost pointer receiver
+    //:: ExpectedOutput(type_error)
+    (*gp).GhostPointerReceiver() // fails because we do not add `gpointer[T]` receivers to method set of T
+    //:: ExpectedOutput(type_error)
+    (*Foo).GhostPointerReceiver(p) // fails because method expects a ghost pointer receiver
+    //:: ExpectedOutput(type_error)
+    (*Foo).GhostPointerReceiver(gp) // fails because `gpointer[T]` receivers are not part of the method set of *T
+    //:: ExpectedOutput(type_error)
+    (Foo).GhostPointerReceiver(p) // fails because method expects a ghost pointer receiver
+    //:: ExpectedOutput(type_error)
+    (Foo).GhostPointerReceiver(gp) // fails because we do not add `gpointer[T]` receivers to method set of T
+    // since `gpointer[Foo]` is a ghost type literal instead of a type literal, we currently do not support parsing the following line:
+    // (gpointer[Foo]).GhostPointerReceiver(p)
+
+    //:: ExpectedOutput(type_error)
+    p.ActualNonPointerReceiver() // actual calls not permitted in ghost code
+    //:: ExpectedOutput(type_error)
+    (*p).ActualNonPointerReceiver() // actual calls not permitted in ghost code
+    //:: ExpectedOutput(type_error)
+    (*Foo).ActualNonPointerReceiver(p) // actual calls not permitted in ghost code
+    //:: ExpectedOutput(type_error)
+    (Foo).ActualNonPointerReceiver(p) // actual calls not permitted in ghost code
+    //:: ExpectedOutput(type_error)
+    (Foo).ActualNonPointerReceiver(*p) // actual calls not permitted in ghost code
+
+    p.GhostNonPointerReceiver() // p is implicitly dereferenced
+    gp.GhostNonPointerReceiver() // gp is implicitly dereferenced
+    (*p).GhostNonPointerReceiver()
+    (*gp).GhostNonPointerReceiver()
+    (*Foo).GhostNonPointerReceiver(p)
+    //:: ExpectedOutput(type_error)
+    (*Foo).GhostNonPointerReceiver(gp) // gp is not of type *Foo
+    //:: ExpectedOutput(type_error)
+    (*Foo).GhostNonPointerReceiver(*gp) // ghost Foo is not assignable to *Foo
+    //:: ExpectedOutput(type_error)
+    (Foo).GhostNonPointerReceiver(p) // *Foo is not assignable to Foo
+    //:: ExpectedOutput(type_error)
+    (Foo).GhostNonPointerReceiver(gp) // gpointer[Foo] is not assignable to Foo
+    (Foo).GhostNonPointerReceiver(*p)
+    (Foo).GhostNonPointerReceiver(*gp)
+}
+
+ghost
+decreases
+preserves acc(r)
+func (r gpointer[Foo]) GhostPointerReceiver() int {
+    r.field = 0
+    return r.field
+}
+
+decreases
+preserves acc(r)
+func (r *Foo) ActualPointerReceiver() int {
+    r.field = 0
+    return r.field
+}
+
+ghost
+decreases
+func (r Foo) GhostNonPointerReceiver() int {
+    r.field = 42
+    return r.field
+}
+
+decreases
+func (r Foo) ActualNonPointerReceiver() int {
+    r.field = 42
+    return r.field
+}
+
+decreases
+preserves acc(r)
+//:: ExpectedOutput(type_error)
+func (r gpointer[Foo]) ActualGhostPointerReceiver() int { // ghostpointer is only allowed for ghost methods
```

Comment:
gpointer is not allowed as the receiver, right? It could be a standard argument?

Reply:
Yes and yes. The parameter would however be explicitly marked as a ghost parameter.